File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

bool cmp(string &a,string &b) {
    
    string tempA=a+b;
    string tempB=b+a;
    return tempA>tempB;
}
string solution(vector<int> numbers) {
    string answer = "";
    vector<string> temp;
    for(int i=0;i<numbers.size();i++) {
        temp.push_back(to_string(numbers[i]));
    }
    
    sort(temp.begin(),temp.end(),cmp);
    for(string a:temp) {
        answer+=a;
    }
    if(answer[0]=='0') {
        return "0";
    }
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    
    
    public String solution(int[] numbers) {
        String answer = "";
        
        List<String> arr = new ArrayList<>();
        for(int i=0;i<numbers.length;i++) {
            arr.add(Integer.toString(numbers[i]));
        }
        
        Collections.sort(arr,new Comparator<String>() {
            
            @Override
            public int compare(String a,String b) {
                String tempA=a+b;
                String tempB=b+a;
                return tempB.compareTo(tempA);
            }
        });
        
        for(String temp:arr) {
            answer+=temp;
        }
        
        if(answer.charAt(0)=='0') {
            return "0";
        } else {
            return answer;
        }
    }
}